lib/player/player.js

```diff
@@ -736,6 +741,36 @@ shaka.player.Player.prototype.setPlaybackRate = function(rate) {
 
 
 /**
+ * @param {Object} restrictions An Object containing the restrictions
+ *     on video tracks that you wish to set. For example, passing
+ *     {'maxBandwidth': 700000, 'minBandwidth': 200000} will restrict
+ *     playback to video tracks with bandwidth between 700000 and 200000.
+ * @export
+ */
+shaka.player.Player.prototype.setRestrictions = function(restrictions) {
+  for (var key in restrictions) {
```

Comment:
Instead of iterating through the keys and updating the internal Restrictions object, let's take an instance of Restrictions from the caller and just replace ours with that.  You'll have pass the new Restrictions to the EmeManager as well.

Then you can replace the 'struct' annotation in Restrictions.  (A struct, in Closure, means that arbitrary properties cannot be added to an instance.  It helps catch typos like "restrictions.mxBandwidth = 5", which would otherwise cause no error since you could well intend to add a new property to this instance.)